web_src/js/features/repo-issue-list.js

```diff
@@ -178,9 +178,44 @@ function initIssuePinSort() {
   });
 }
 
+function initIssueListLabelFilter() {
+  function excludeLabel(item) {
+    const href = $(item).attr('href');
+    const id = $(item).data('label-id');
+
+    const regStr = `labels=((?:-?[0-9]+%2c)*)(${id})((?:%2c-?[0-9]+)*)&`;
+    const newStr = 'labels=$1-$2$3&';
+
+    window.location = href.replace(new RegExp(regStr), newStr);
+  }
```

Comment:
I think it's better to move this function to top-level. There is a [linter rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/consistent-function-scoping.md) for this, but it appears we have not enabled it.

Reply:
Interestingly, we do have the rule enabled:

https://github.com/go-gitea/gitea/blob/86d019682f0406e6e1739a31d72a2686ff8764cb/.eslintrc.yaml#L627

Reply:
Actually I do not want to move it to public scope. This function is a private function, it is only used by this `initIssueListLabelFilter`.

And, these code blocks are moved from old code, so I would like to keep the same structure as a refactoring move.

Reply:
Keeping such function nested has multiple drawbacks:

- It makes the function untestable because it can not be exported
- It prevents the JS engine from opimizing because the function is re-created every time the parent calls
- It is less readable

Reply:
If you really want to keep it in this place, use a function expression (`const excludeLabel = () => {}`) to make the indent clear.

Reply:
Actually, no need to touch the old code.

Now `+3 −2`